packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -126,52 +155,93 @@ const dialPadButtonsDefault: _DialpadButtonProps[][] = [
+  styles?: DialpadStyles;
 }): JSX.Element => {
   const theme = useTheme();
   const [textValue, setTextValue] = useState('');
   const [error, setError] = useState('');
-  const onClickDialpad = (input: string): void => {
-    setError('');
 
+  const { onSendDtmfTone, onClickDialpadButton, onDisplayDialpadInput } = props;
+
+  const onClickDialpad = (input: string, index: number): void => {
+    setError('');
     setTextValue(textValue + input);
+    if (onSendDtmfTone) {
+      onSendDtmfTone(DtmfTones[index]);
+    }
+    if (onClickDialpadButton) {
+      onClickDialpadButton();
+    }
   };
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const setText = (e: any): void => {
-    // if text content includes non-valid dialpad input return an error
-    if (!`${e.target.value}`.match(/^[0-9\s*#+()-]*$/)) {
-      setError(
-        _formatString(props.errorText, { invalidCharacter: `${e.target.value.slice(e.target.value.length - 1)}` })
-      );
-    } else {
-      setError('');
-      setTextValue(e.target.value);
-    }
+    setError('');
+    setTextValue(e.target.value);
   };
 
-  const dialpadButtonsContent = props.dialpadButtons ?? dialPadButtonsDefault;
+  // const dialpadButtonsContent = props.dialpadButtons ?? dialPadButtonsDefault;
 
   return (
-    <div className={mergeStyles(containerStyles(theme), props.styles?.root)}>
+    <div className={mergeStyles(containerStyles(theme), props.styles?.root)} id="dialpadContainer">
       <TextField
         styles={concatStyleSets(textFieldStyles(theme), props.styles?.textField)}
-        value={formatPhoneNumber(textValue)}
+        value={onDisplayDialpadInput ? onDisplayDialpadInput(textValue) : formatPhoneNumber(textValue)}
         onChange={setText}
         errorMessage={error}
         placeholder={props.defaultText}
+        id="dialpad-input"
       />
       <FocusZone>
-        {dialpadButtonsContent.map((rows, i) => {
+        {dialPadButtonsDefault.map((rows, index) => {
           return (
-            <Stack horizontal key={`row_${i}`} horizontalAlign="stretch">
+            <Stack horizontal key={`row_${index}`} horizontalAlign="stretch">
               {rows.map((button, i) => (
                 <DialpadButton
                   key={`button_${i}`}
+                  /* row index = 0
+                  i: (0,1,2) => (0,1,2)
+                  row index = 1
+                  i: (0,1,2)=> (3,4,5)
+                  row index = 2
+                  i: (0,1,2)=> (6,7,8)
+                  row index = 3
+                  i: (0,1,2)=> (9,10,11)
+                  i + index*rows.length calculates the corresponding index for each button
+                  dialpad index:
+                  0 1 2
+                  3 4 5
+                  6 7 8
+                  9 10 11
+                  then use this index to locate the corresponding dtmf tones
+                  DtmfTones[index]
+                  */
+                  index={i + index * rows.length}
```

Comment:
don't use `i` as well as `index`, `i` means `index` and these are two different variables. Instead use `rowIndex` and `columnIndex` or `x` and `y` or even `i` and `j`

Reply:
fixed that!